fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -355,7 +377,17 @@ public MaterializedView selectiveCopy(Collection<String> reservedPartitions, boo
     @Override
     public void gsonPostProcess() throws IOException {
         super.gsonPostProcess();
+    }
 
+    @Override
+    public void write(DataOutput out) throws IOException {
+        // write type first
+        Text.writeString(out, type.name());
```

Comment:
Because read type first when reading table：
```
// Table 229
public static Table read(DataInput in) throws IOException {
        Table table = null;
        TableType type = TableType.valueOf(Text.readString(in));
```